### PR TITLE
DSERV-597-genes-structure

### PR DESCRIPTION
**Pull request by mingjiecn**

### CODE REVIEW

File: src/routers/datatypeRouters/nodes/genes_structure.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  if (!fromProtein) {
+    const query = `
+      FOR record in ${genesStructureSchema.db_collection_name as string}
+      ${filterBy}
+      SORT record.gene_id, record.transcript_id, record['start:long']
+      LIMIT ${page as number * limit}, ${(page as number + 1) * limit}
+      RETURN {${getDBReturnStatements(genesStructureSchema)}}
+    `
+    return await (await db.query(query)).all()
+  } else {
```
Comment (shengchd) [inline src/routers/datatypeRouters/nodes/genes_structure.ts:102]: Am I understanding correctly that the protein query fields will override other fields then? Maybe it's fine, we can just make it clear in the endpoint description part.

File: src/routers/datatypeRouters/nodes/genes_structure.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    const proteinInput = {
+      protein_id: input.protein_id,
+      protein_name: input.protein_name,
+      organism: input.organism,
+      page: 0,
+      limit: 25
```
Comment (shengchd) [inline src/routers/datatypeRouters/nodes/genes_structure.ts:108]: I'm wondering how this number of limit is set?

**Review by shengchd — COMMENTED** [review]
Looks good overall, just have two questions on the protein query part.

File: src/routers/datatypeRouters/nodes/genes_structure.ts (hunk 1 above)
Comment (mingjiecn) [reply src/routers/datatypeRouters/nodes/genes_structure.ts:102]: I updated the description.

File: src/routers/datatypeRouters/nodes/genes_structure.ts (hunk 2 above)
Comment (mingjiecn) [reply src/routers/datatypeRouters/nodes/genes_structure.ts:108]: actually the limit is already set in the query function, so I will remove it. Thanks!

**Review by shengchd — APPROVED** [review]
looks good

**Pull request merged**